### DISCUSSION TITLE
[error] [http client] failed to uncompress content starting with:

### DISCUSSION THREAD

**Discussion by jcl-64** [glpi-project/glpi-agent#335] [Q&A]
### Bug reporting acknowledgment

Yes, I read it

### Professional support

I still have but want a public following to keep the community aware

### Describe the bug

in glpi-agent.log file, closing each inventory report, last sentence is always : 
**[error] [http client] failed to uncompress content starting with:**
(There is nothing else after that)

- Computer inventory is done and glpi server receive it with all components 
- but there is no result for the tasks "network scan NMAP"
- 5 agents on 5 computers give the same error
- installed agents may scan the netwok when using the toolbox agent
- All the tasks and jobs "Scan NMAP" fail with message "The agent requests a configuration that has already been sent to it by the server. The agent may have encountered a critical error"

### To reproduce

glpi 10.0.5 on Ubuntu
plug'in GLPI Inventory
Agent 1.4

### Expected behavior

waiting for "network scan" task complete to add network components

### Operating system

Linux

### GLPI Agent version

1.4

### GLPI version

10.0.5

### GLPIInventory plugin or FusionInventory for GLPI plugin version

GLPI Inventory v1.1.0

### Additional context

Many thanks if someone have had the same issue and have an answer.

**Comment by stonebuzz** [comment]
SNMP Discovery and Inventory are not design for Computer 
It's for Network Equipment (Switch, Printer).
Please install agent on it.

Best regards

**Reply by jcl-64** [reply]
Thank you for answer. Computer discovery work well with agent installed on computer and computers appear in GLPI.
I now want to discover Network Equipment with SNMP discovery & Inventory and add them in GLPI.
It works from the agent 1.4 : localhost:62534/toolbox show me ip adresses of switchs & printers. But no way to see them in GLPI : none obejct is created. 
agent 1.4 logs end with sentence "[error] [http client] failed to uncompress content starting with:"

**Reply by g-bougard** [reply]
Hi @jcl-64 
agent toolbox will only generate XML locally. But you can use glpi-injector to upload generated XML to the server.
But you can also enable GlpiInventory plugin from the GLPI Marketplace to handle netdiscovery & netinventory tasks from GLPI. Then be careful to setup your agent server url as explained here: https://glpi-agent.readthedocs.io/en/latest/configuration.html#server

**Reply by jcl-64** [reply]
Thanks for answer.
I tried Toolbox to be sure agent can find SNMP network equipement : ok it works
If i try to manually import XML files : ok it works
I have enabled GlpiInventory plugin from the GLPI Marketplace and careful setup everything (with server = "https://myserver/marketplace/glpiinventory").

The issue is that GLPI task "discover network" doesn't end. It turns for hours even if range IP is only 3 IP
From GLPI, no errors in "php-erros.log" but "pluginGlpiinventory-tasks.log" is now 31 911 504 bites !!!
From the agent, always ----> "[error] [http client] failed to uncompress content starting with:"
then chinese characters -> see glpi-agent.log
[glpi-agent.log](https://github.com/glpi-project/glpi-agent/files/10700222/glpi-agent.log)

2,5 all days to try everything... I don't understand 
Thnak you for your help

**Reply by jcl-64** [reply]
New : in php-errors.log, i can read : 
[2023-02-09 19:02:06] glpiphplog.WARNING:   *** PHP Warning (2): Undefined array key "name" in /var/www/myserver/src/Agent.php at line 253
  Backtrace :
  src/Search.php:7451                                Agent::getSpecificValueToDisplay()
  src/Search.php:1701                                Search::giveItem()
  src/Search.php:448                                 Search::constructData()
  src/Search.php:200                                 Search::getDatas()
  src/Search.php:177                                 Search::showList()
  front/agent.php:42                                 Search::show()

**Reply by jcl-64** [reply]
and then nothing append during then task... 40 mn waiting, never seen something green
![task](https://user-images.githubusercontent.com/67611204/217901244-c4cbb8a3-f4dd-482f-8d84-64b4f9ca983c.JPG)

**Reply by g-bougard** [reply]
> Thanks for answer. I tried Toolbox to be sure agent can find SNMP network equipement : ok it works If i try to manually import XML files : ok it works I have enabled GlpiInventory plugin from the GLPI Marketplace and careful setup everything (with server = "https://myserver/marketplace/glpiinventory").
> 
> The issue is that GLPI task "discover network" doesn't end. It turns for hours even if range IP is only 3 IP From GLPI, no errors in "php-erros.log" but "pluginGlpiinventory-tasks.log" is now 31 911 504 bites !!! From the agent, always ----> "[error] [http client] failed to uncompress content starting with:" then chinese characters -> see glpi-agent.log [glpi-agent.log](https://github.com/glpi-project/glpi-agent/files/10700222/glpi-agent.log)
> 
> 2,5 all days to try everything... I don't understand Thnak you for your help

Please enable `debug = 2` on the agent: you can set it in the registry into `HK_LM/Software/GLPI-Agent` & restart the service. Then force an inventory or wait for it.

**Reply by jcl-64** [reply]
News : with a new GLPI, not https://myserver but http://localhost/glpi, same config, same agent, everything works well : task nmap and snmp are green !! 
I continue my research...

**Reply by jcl-64** [reply]
debug = 2  + registry into HK_LM/Software/GLPI-Agent -> I did, very convenient to try as often you need
But no solution for the moment

**Reply by g-bougard** [reply]
What's the differences between the first and second GLPI server ?
This seems then to be a server configuration problem, maybe the config of the httpd service itself.

**Reply by jcl-64** [reply]
first server is a production server, remote at OVh, on Ubuntu and SSL : everything works well for 70 users
second server is localhost, for test, with just http 
As i said, 

TESTS 
1/ on the production server, with glpiinventory : 
For computer inventorys -> it's ok
For network discover tasks -> not ok, tasks start from GLPI when local agent start, but nothing append, they never finish
2/ on the glpi test server, with glpiinventory (same agent, same computer) : everything works
3/ on the production server, with Agent Fusion Inventory 2.6 & Fusion Inventory Plugin : everything works at the first time !

ERROR when 1/
From production server, i can't find any error
From GLPI Agent 1.4 (and 1.5 nightly), the only bug i see is in glpi-agent.log : "[error] [http client] 927A86A3: failed to uncompress content starting with:  xœ«VJ­(È,J,ÉÌÏS²R22QÒQ*.I "

**Reply by g-bougard** [reply]
So this is probably a SSL support configuration issue on your OVH server. If it's working better with legacy softwares like FI, maybe your SSL support layer on your httpd daemon is not well configured. The "failed to uncompress" looks like the http client failed to decrypt the SSL messages. To me, this is just a symptom something is wrong in your server configuration. Do you use a publicly sign certificate or a self-sign one ?

**Reply by jcl-64** [reply]
I use publicy certificate Let's Encrypt, my certificat provider is https://r3.i.lencr.org/. Nothing in my configuration seems to me to be faulty except maybe my certificates has fields Organisation (O) & (OU) empty.
Do not forget either that with Fusion Inventory everything works very well on this same server and this SSL config 

If i focus on glpi-agent.log messages, : 
With localhost, i can see many message like "[debug2] [http client] 9504493D: received message:" and ...
With production server, no messages contain "received message:"

I now imagine testing by removing the https on my production server ; a bit hard apache config ... I'll tell you later

**Reply by g-bougard** [reply]
You may also try the glpi-agent nightly build in the case something changed in the SSL & HTTP client support.

**Reply by jcl-64** [reply]
1/ I tried glpi-agent nightly -> same behavior

2/ Then, i create a new instance on my server "empty_glpi" : same server + same sql instance + same SSL certificates + new database -> glpi -agent work ! 

3/ Then, i restore my database production into "empty_glpi" :  same server + same sql instance + same SSL certificates + same database -> and.... glpi -agent work !

But there is something wrong when I restore my database -> glpi entity parent "root" lost his daugters !!!
With Super-Admin, récursif, with view root (tree) : 
-> .../front/entity.php show me only root entity
-> .../front/entity.php/entites TAB show me all the entities but i can't click on them
-> .../front/user.php show me only users of root entity
BUT
I can log in any entity and work as usual

**Reply by jcl-64** [reply]
Finally, after 5 days with this issue, I can not find from where this anomaly "[error] [http client] 927A86A3: failed to uncompress content starting with:" comes from. 
Reinstalling from scratch and remounting the backup miraculously fixed the problem for our main instance, but not on a secondary instance. To be resumed later ! In the meantime if anyone has this error, we can exchange...

**Comment by g-bougard** [comment]
Hi @jcl-64 
I converted your issue to & Q&A discussion as you seem to first need some help to understand the problem and you don't provide any info that could indicate there's a bug in glpi-agent.
Even, I don't understand why you're using "NMAP" in task name as glpi-agent doesn't use it. This leaves me think you're misunderstanding something. And as @stonebuzz said, netdiscovery & netinventory tasks can only help to inventory network devices & printers with snmp support.

**Reply by jcl-64** [reply]
News : 
With Agent Fusion Inventory 2.6 -> GLPI 10.0.6 Fusion Inventory Plugin : every works at the first time !
![image](https://user-images.githubusercontent.com/67611204/218041240-83ff8f6f-b730-4178-bb6f-03ff4646fd9f.png)
